server/src/main/scala/services/crunch/deskrecs/DynamicRunnableDeskRecs.scala

```diff
@@ -0,0 +1,152 @@
+package services.crunch.deskrecs
+
+import actors.acking.AckingReceiver.{Ack, StreamCompleted, StreamFailure, StreamInitialized}
+import akka.NotUsed
+import akka.actor.ActorRef
+import akka.stream.ClosedShape
+import akka.stream.scaladsl.GraphDSL.Implicits
+import akka.stream.scaladsl.{GraphDSL, RunnableGraph, Sink, Source}
+import drt.shared.CrunchApi.{DeskRecMinutes, MillisSinceEpoch}
+import drt.shared.FlightsApi.FlightsWithSplits
+import drt.shared.{ApiFlightWithSplits, ArrivalKey, SDateLike}
+import drt.shared.Terminals.Terminal
+import drt.shared.api.Arrival
+import manifests.queues.SplitsCalculator
+import manifests.queues.SplitsCalculator.SplitsForArrival
+import org.slf4j.{Logger, LoggerFactory}
+import passengersplits.parsing.VoyageManifestParser.{VoyageManifest, VoyageManifests}
+import services.SDate
+import services.crunch.desklimits.TerminalDeskLimitsLike
+
+import scala.collection.immutable.Map
+import scala.concurrent.{ExecutionContext, Future}
+
+object DynamicRunnableDeskRecs {
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  def createGraph(deskRecsSinkActor: ActorRef,
+                  daysSourceQueue: Source[MillisSinceEpoch, NotUsed],
+                  daysToDeskRecs: DaysToDeskRecs)
+                 (implicit ec: ExecutionContext): RunnableGraph[NotUsed] = {
+
+    val deskRecsSink = Sink.actorRefWithAck(deskRecsSinkActor, StreamInitialized, Ack, StreamCompleted, StreamFailure)
+
+    val graph = GraphDSL.create(daysSourceQueue) {
+      implicit builder =>
+        daysSourceQueueAsync =>
+          daysToDeskRecs(daysSourceQueueAsync.out) ~> deskRecsSink
+          ClosedShape
+    }
+
+    RunnableGraph.fromGraph(graph)
+  }
+
+  type DaysToDeskRecs = Implicits.PortOps[MillisSinceEpoch] => Implicits.PortOps[DeskRecMinutes]
+
+  def daysToDeskRecs(flightsProvider: MillisSinceEpoch => Future[Source[FlightsWithSplits, NotUsed]],
+                     liveManifestsProvider: MillisSinceEpoch => Future[Source[VoyageManifests, NotUsed]],
+                     historicManifestsProvider: Iterable[Arrival] => Future[Map[ArrivalKey, VoyageManifest]],
+                     crunchStart: SDateLike => SDateLike,
+                     splitsCalculator: SplitsCalculator,
+                     portDeskRecs: DesksAndWaitsPortProviderLike,
```

Comment:
This would be less confusing if it was called `PortDesksAndWaitsProviderLike`

Reply:
Renamed! And also renamed the terminal version